src/acme/ui/CustomerCreatePanel.java

```diff
@@ -0,0 +1,144 @@
+    }
+	
+	@Override
+    public void buildPanel() {
+        company = this.getCompany();
+        this.removeAll();
+        
+        setLayout(null);
+        
+        JLabel titleLabel = new JLabel("Customer");
+        titleLabel.setBounds(42, 30, 75, 22);
+        titleLabel.setFont(new Font("Tahoma", Font.PLAIN, 18));
+        add(titleLabel);
+        
+        JLabel nameLabel = new JLabel("Name");
+        nameLabel.setBounds(57, 88, 46, 14);
+        add(nameLabel);
+        
+        nameField = new JTextField();
+        nameField.setBounds(113, 85, 125, 20);
+        add(nameField);
+        nameField.setColumns(10);
+        
+        JLabel numberLabel = new JLabel("Number");
+        numberLabel.setBounds(57, 126, 46, 14);
+        add(numberLabel);
+        
+        
+        numberField = new JLabel(Integer.toString(Customer.getNextCustomerNumber()));
+        numberField.setBounds(113, 126, 46, 14);
+        add(numberField);
+        
+        JLabel lblIntersection = new JLabel("Intersection");
+        lblIntersection.setFont(new Font("Tahoma", Font.PLAIN, 14));
+        lblIntersection.setBounds(57, 166, 75, 22);
+        add(lblIntersection);
+        
+        JLabel avenueLabel = new JLabel("Avenue");
+        avenueLabel.setBounds(57, 199, 46, 14);
+        add(avenueLabel);
+        
+        JLabel streetLabel = new JLabel("Street");
+        streetLabel.setBounds(57, 240, 46, 14);
+        add(streetLabel);
+        
+        Set<Road> avenues = company.getMap().getAvenues();
+        avenueComboField = new JComboBox<Road>(avenues.toArray(new Road[0]));
+        avenueComboField.setBounds(113, 196, 125, 20);
+        add(avenueComboField);
+        
+        Set<Road> streets = company.getMap().getStreets();
+        streetComboField = new JComboBox<Road>(streets.toArray(new Road[0]));
+        streetComboField.setBounds(113, 237, 125, 20);
+        add(streetComboField);
+        
+        JButton saveButton = new JButton("Save");
+        saveButton.addActionListener(e -> saveCustomer());
+        saveButton.setBounds(330, 266, 89, 23);
+        add(saveButton);
+        
+        JButton cancelButton = new JButton("Cancel");
+        cancelButton.addActionListener(e -> cancelCreate());
+        cancelButton.setBounds(231, 266, 89, 23);
+        add(cancelButton);
+        
+        this.revalidate();
+        this.repaint();
+	}
+	
+	private void navigateToPreviousPage() {
+		this.getAcmeUI().courierList();
+	}
+	
+	private void cancelCreate() {
+		customer.delete();
+		navigateToPreviousPage();
+	}
+	
+	private void updateCustomerFromFields() {
+		customer.setName(nameField.getText());
+		customer.setCustomerNumber(Integer.parseInt(numberField.getText()));
+		Road avenue = (Road) avenueComboField.getSelectedItem();
+		Road street = (Road) streetComboField.getSelectedItem();
+		customer.setAvenueName(avenue.getName());
+		customer.setStreetName(street.getName());
+		customer.setActive(true);

```

Comment:
It looks like you never set the Intersection. The intersection is not assumed from setting the street/avenue.

Reply:
Now we calculate the intersection each time we get it. See [here](https://github.com/simpleman19/acmeDSTS/pull/23/commits/2318893e87a5cf592824f2393c0e5509e9c046bc) where I changed `customer.getIntersection()` (@simpleman19, you can cherry-pick this commit for Database seed)